tasks/interview-corejs.md

```diff
@@ -1,51 +1,86 @@
-### Question examples:
-   * Data types in JS. Type coercion.
-   * What is hoisting?
-   * Let vs var. Const.
-   * Passing data by value and by reference. Examples.
-   * `{a: 10} == {a: 10}`. What will the code return?
-   * What is `this`?
-   * Apply, call, bind. What are they used for? What are the differences?
-   * Closure. Give an example.
-   * Array methods that loop over the elements.
-   * Object methods to iterate over the elements
-   * Map, Set, WeakMap, WeakSet
-   * Browser events. Browser default actions abortion.
-   * Event bubbling and event capturing.
-   * Event delegation. Example.
-   * Event loop. Explain how stack and two queues communicate with each other in the loop?
-   * Promises
-   * spread/rest
-   * Any other questions on CoreJS/DOM/Event Loop/Inheritance/etc.
-                         
-### Coding tasks:
-* `Function.prototype.bind` implement polyfill
-* `Object.create` implement polyfill
-* `Array.flat` implement polyfill
-* 'hello world'.repeating(3) -> 'hello world hello world hello world'. How to implement?
-*  myFunc('!', 4, -10, 34, 0) -> '4!-10!34!0`. How to implement?
-*  five(plus(seven(minus(three())))) -> 9. How to implement?
-*  add(5)(9)(-4)(1) -> 11. How to implement?
-* `periodOutput(period)` method should output in the console once per every period how much time left from the function first call. 
+## JS Questions:
+
+- ### Objects Built-in methods.
+
+  - Know how to use built-in methods
+  - Know static Object methods
+  - Property flags & descriptors (student is able to set property via Object. defineProperty)
+
+- ### Object as Hash.
+
+  - Be able to loop through Object keys
+  - Know how to create iterable objects, Symbol.iterator usage
+
+- ### Arrays Built-in methods
+
```

Comment:
How to determine if object is Array (I'm not sure is it for this section or for somewhere else)

Reply:
moved to another section